be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -346,7 +345,13 @@ Status BetaRowsetWriter::_flush_segment_writer(std::unique_ptr<segment_v2::Segme
     DCHECK_LE(min_key.compare(max_key), 0);
     key_bounds.set_min_key(min_key.to_string());
     key_bounds.set_max_key(max_key.to_string());
-    _segments_encoded_key_bounds.emplace_back(key_bounds);
+    {
+        std::lock_guard<SpinLock> l(_lock);
+        _segment_num_rows.reserve(_num_segment);
```

Comment:
should use `resize()` here?

Reply:
emm... right
I want to use `resize(segment_id)` at first, but resize may shrink the vector size, so changed to `reserve(segment_id)`, but ignored the fact that `reserve` won't change size of vector.